### PR TITLE
mbedtls_mpi_sub_abs: Skip memcpy when redundant (#6701).

### PR DESCRIPTION
**Pull request by ucko**
## Description

In some contexts, the output pointer may equal the first input pointer, in which case copying is not only superfluous but results in "Source and destination overlap in memcpy" errors from Valgrind (as I observed in the context of ecp_double_jac) and a diagnostic message from TrustInSoft Analyzer (as Pascal Cuoq reported in the context of other ECP functions called by cert-app with a suitable certificate).

I will be happy to supply a corresponding LTS PR in due course, particularly given that I'm using that branch myself.

(Split from #6930.)

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** #7001
- [x] **tests** not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: ChangeLog.d/conditionalize-mbedtls_mpi_sub_abs-memcpy.txt
Comment (tom-cosgrove-arm) [inline ChangeLog.d/conditionalize-mbedtls_mpi_sub_abs-memcpy.txt:5]: I think the first sentence could be simply

`Fix potential undefined behaviour in mbedtls_mpi_sub_abs ().`

(ChangeLog entries are user facing, and users probably don't need to know that level of detail)

**Comment by ucko** [comment]
That's fair, though I'll go with the US spelling of "behavior", for which I do see precedent. ;-)

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Looks good to me, but please [rewrite the commit messages to ensure that all commits have a signoff line](https://github.com/Mbed-TLS/mbedtls/pull/6942/checks?check_run_id=10708747165).

**Comment by ucko** [comment]
AFAICT, only the ChangeLog simplification lacked such a line; fixed.  Thanks, and sorry for the accidental omission.

**Comment by tom-cosgrove-arm** [comment]
Thanks for fixing that; can you prepare the backport now? We normally merge a PR and its backport at the same time, so we need both before this can be merged.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM; just needs CI (I don't expect any problems now) and the backport

**Comment by ucko** [comment]
Sounds good, thanks!  I've pushed a squashed backport: https://github.com/ucko/mbedtls/tree/2023a-bignum-2.28

**Comment by tom-cosgrove-arm** [comment]
The backport should be raised as a second PR, targeting the branch `mbedtls-2.28`, with the same title as the original but with `[Backport 2.28]` prepended to the title. See, for example, PRs #6982 and #6983

**Comment by ucko** [comment]
So I see, thanks.  Opened #7001.

**Review by gilles-peskine-arm — APPROVED** [review]
Thanks, LGTM

**Pull request merged**
